core/cas-server-core-tickets-api/src/main/java/org/apereo/cas/ticket/registry/AbstractTicketRegistry.java

```diff
@@ -58,14 +59,27 @@ public void addTicket(final Ticket ticket) throws Exception {
 
     @Override
     public Ticket getTicket(final String ticketId) {
-        return getTicket(ticketId, ticket -> {
+        Ticket returnTicket = getTicket(ticketId, ticket -> {
             if (ticket == null || ticket.isExpired()) {
-                LOGGER.debug("Ticket [{}] has expired and will be removed from the ticket registry", ticketId);
+                if (ticket == null) {
+                    LOGGER.debug("Ticket [{}] not found but will be removed from the ticket registry just in case", ticketId);
+                } else {
+                    val ticketAgeSeconds = ZonedDateTime.now(ticket.getExpirationPolicy().getClock()).toEpochSecond() - ticket.getCreationTime().toEpochSecond();
```

Comment:
I made a private method to avoid the duplication. I didn't see a spot in the method to do the code and didn't think it was warranted in the Ticket interface, at least not for this.